Sources/Filters/Core/TransformPolyDataFilter/index.js

```diff
@@ -0,0 +1,201 @@
+        inputPointsData[i * 3 + 1],
+        inputPointsData[i * 3 + 2],
+      ];
+
+      model.transform.transformPoint(point, transformedPoint);
+      outputPointsData[i * 3] = transformedPoint[0];
+      outputPointsData[i * 3 + 1] = transformedPoint[1];
+      outputPointsData[i * 3 + 2] = transformedPoint[2];
+    }
+
+    // Transform vectors if present
+    let outputVectors = null;
+    if (inputVectors) {
+      outputVectors = vtkDataArray.newInstance({
+        name: inputVectors.getName(),
+        numberOfComponents: 3,
+        dataType: VtkDataTypes.FLOAT,
+        values: new Float32Array(numPoints * 3),
+      });
+
+      const inputVectorsData = inputVectors.getData();
+      const outputVectorsData = outputVectors.getData();
+      const transformedVector = [];
+
+      for (let i = 0; i < numPoints; i++) {
+        const vector = [
+          inputVectorsData[i * 3],
+          inputVectorsData[i * 3 + 1],
+          inputVectorsData[i * 3 + 2],
+        ];
+        model.transform.transformVector(vector, transformedVector);
+        outputVectorsData[i * 3] = transformedVector[0];
+        outputVectorsData[i * 3 + 1] = transformedVector[1];
+        outputVectorsData[i * 3 + 2] = transformedVector[2];
+      }
+    }
+
+    // Transform normals if present
+    let outputNormals = null;
+    if (inputNormals) {
+      outputNormals = vtkDataArray.newInstance({
+        name: inputNormals.getName(),
+        numberOfComponents: 3,
+        dataType: VtkDataTypes.FLOAT,
+        values: new Float32Array(numPoints * 3),
+      });
+
+      const inputNormalsData = inputNormals.getData();
+      const outputNormalsData = outputNormals.getData();
+      const transformedNormal = [];
+
+      for (let i = 0; i < numPoints; i++) {
+        const normal = [
+          inputNormalsData[i * 3],
+          inputNormalsData[i * 3 + 1],
+          inputNormalsData[i * 3 + 2],
+        ];
+
+        model.transform.transformNormal(normal, transformedNormal);
+        outputNormalsData[i * 3] = transformedNormal[0];
+        outputNormalsData[i * 3 + 1] = transformedNormal[1];
+        outputNormalsData[i * 3 + 2] = transformedNormal[2];
+      }
+    }
+
+    // TODO: Transform cell vectors and normals if transform is linear
+
+    // Set output data
+    output.setPoints(outputPoints);
+
+    // Copy cell topology
+    output.setVerts(input.getVerts());
+    output.setLines(input.getLines());
+    output.setPolys(input.getPolys());
+    output.setStrips(input.getStrips());
+
+    // Set transformed point data
+    if (outputNormals) {
+      outputPointData.setNormals(outputNormals);
+    }
+    if (outputVectors) {
+      outputPointData.setVectors(outputVectors);
+    }
+
+    // Pass through other data
+    outputPointData.passData(inputPointData);
+
+    return true;
+  }
+
+  publicAPI.requestData = (inData, outData) => {
+    const input = inData[0];
+    const output = vtkPolyData.newInstance();
```

Comment:
if `output[0]` already exists, it shouldn't allocate vtkPolyData again

Reply:
Done